common/os/os.cxx

```diff
@@ -66,10 +68,25 @@ static const char* gethomedir(bool userDir)
   if (userDir)
     return homedir;
 
+  // check if (deprecated) legacy path exists and use that if so
   snprintf(dir, sizeof(dir), "%s/.vnc", homedir);
+  struct stat st;
+
+  if (stat(dir, &st) == 0)
+    return dir;
+
+  if (xdg_def != NULL) {
+    xdgdir = getenv(xdg_env);
+    if (xdgdir != NULL && xdgdir[0] == '/')
+      snprintf(dir, sizeof(dir), "%s/tigervnc", xdgdir);
+    else
+      snprintf(dir, sizeof(dir), "%s/%s/tigervnc", homedir, xdg_def);
+  }
 
   return dir;
 #else
+  UNUSED(xdg_def);
+  UNUSED(xdg_env);
```

Comment:
We haven't used a macro like this in other places, so just inline things to keep everything consistent.